src/backend/joanie/core/api/client.py

```diff
@@ -1049,6 +1057,86 @@ def download(self, request, pk=None):  # pylint: disable=no-self-use, unused-arg
             filename=f"{contract.definition.title}.pdf".replace(" ", "_"),
         )
 
+    @action(
+        methods=["GET"],
+        detail=False,
+        url_name="zip_archive",
```

Comment:
```suggestion
        url_name="zip-archive",
```
Seems cleaner to me.